### PR TITLE
Reland 3 of Merge more symbolic meta kernels and symint changes from branch

### PR DESCRIPTION
**Pull request by albanD**
Take 3
Contains:
- symintification of split*
- floor support on SymFloat
- pad_backward, gather, scatter meta

### CODE REVIEW

**Comment by z-a-f** [comment]
Just gonna add some context here :)

https://docs.google.com/document/d/1iiLNwR5ohAsw_ymfnOpDsyF6L9RTUaHMpD8YLw-jxEw/edit

File: torch/_meta_registrations.py [hunk 1]
```diff
@@ -1257,6 +1325,187 @@ def meta_slice_scatter(self, src, dim=0, start=None, end=None, step=1):
...
+@register_meta(aten.scatter_add.default, register_dispatcher=False)
+def meta_scatter_add(self, dim, index, src):
+    scatter_meta_impl(self, dim, index, src, "add")
+    return self.new_empty(self.shape)
+
+
+@register_meta(aten.upsample_nearest2d.vec)
+def upsample_nearest2d_vec(input, output_size, scale_factors):
+    mem_format = utils.suggest_memory_format(input)
+    spatial_dimensions = input.dim() - 2
```
Comment (z-a-f) [inline torch/_meta_registrations.py:1486]: Should there be an shape check before this?

File: aten/src/ATen/FunctionalInverses.cpp [hunk 2]
```diff
@@ -177,31 +177,31 @@ Tensor FunctionalInverses::slice_copy_Tensor_inverse(const Tensor& base, const T
...
     // on top of the base tensor.
     // For autograd, we have all of the tensors outputted by split() and we just want to stack them.
-    dim = at::maybe_wrap_dim(dim, base.sizes().size());
-    auto dim_size = base.size(dim);
-    auto start = mutated_view_idx * split_size;
-    auto end = start + split_size;
+    dim = at::maybe_wrap_dim(dim, base.dim());
+    auto dim_size = base.sym_size(dim);
+    auto start = split_size * mutated_view_idx;
+    auto end = split_size + start;
```
Comment (z-a-f) [inline aten/src/ATen/FunctionalInverses.cpp:188]: Personal pet peeve? :)

**Review by z-a-f — APPROVED** [review]

File: aten/src/ATen/FunctionalInverses.cpp (hunk 2 above)
Comment (albanD) [reply aten/src/ATen/FunctionalInverses.cpp:188]: Yes that PR was done before we added the radd haha Doesn't change anything.

**Pull request closed without merging**